src/ir/display.rs

```diff
@@ -3,18 +3,47 @@
 use super::{FuncDecl, FunctionBody, Module, SourceLoc, ValueDef};
 use crate::entity::EntityRef;
 use std::collections::HashMap;
-use std::fmt::{Display, Formatter, Result as FmtResult};
+use std::fmt::{self, Display, Formatter, Result as FmtResult};
+
+/// Hooks to print information after inst, before and after blocks
+/// and before and after functions.
+pub trait PrintDecorator {
+    fn after_inst(&self, _value: super::Value, _f: &mut fmt::Formatter) -> fmt::Result {
```

Comment:
It's a tiny bit silly in this case (since the purpose is fairly apparent) but I like to stick to a style of mandatory doc comments on all items that are `pub` -- could we add those here too? Something like:

```
/// Print arbitrary text after an instruction.
///
/// Invoked after every instruction in a block. The instruction has already been printed on its own line; this method can print content on the line below the operator, if desired.
```

and likewise below.

Reply:
For clarity, the `after_inst` instruction prints information on the same line as the operator after the operator has been printed.